daemon/daemon.go

```diff
@@ -155,18 +155,20 @@ type Daemon struct {
 }
 
 // UpdateProxyRedirect updates the redirect rules in the proxy for a particular
-// endpoint using the provided L4 filter. Returns the allocated proxy port
-func (d *Daemon) UpdateProxyRedirect(e *endpoint.Endpoint, l4 *policy.L4Filter, proxyWaitGroup *completion.WaitGroup) (uint16, error) {
+// endpoint using the provided L4 filter. Returns the allocated proxy port via
+// 'acked' callback, which will be called if the operation is successful
+// before the 'proxyWaitGroup' is canceled.
+func (d *Daemon) UpdateProxyRedirect(e *endpoint.Endpoint, l4 *policy.L4Filter, proxyWaitGroup *completion.WaitGroup, acked func(redirectPort uint16), nacked func(error)) error {
```

Comment:
This definitely conflicts with the changes I made in #5451 to finalize (resp. revert) xDS changes on regeneration success (resp. failure).
I would favor removing your changes here to the signature of `UpdateProxyRedirect` and other related methods, because the finalize/revert approach in #5451 is more general as it also covers NACKs on network policy updates and redirect removals.

Reply:
Reverted these changes.